src/Options/Applicative/Simple.hs

```diff
@@ -83,15 +86,18 @@ simpleVersion version =
   [|concat (["Version "
            ,$(TH.lift $ showVersion version)
            ] ++
-           if $gitHash == ("UNKNOWN" :: String)
-             then []
-             else
-               [", Git revision "
-               ,$gitHash
-               ,if $gitDirty
-                   then " (dirty)"
-                   else ""
-               ])|]
+           case giResult of
+             Left _ -> []
+             Right gi ->
+               if giHash gi == "UNKNOWN"
```

Comment:
I _think_ this should be unnecessary now, since `gi` itself will do all error checking and return a `Left` if relevant.